typescript/introduction/tsc-basics/tsc-interfaces-intro.md

```diff
@@ -0,0 +1,69 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Interfaces Intro
+
+---
+## Content
+
+One of the most popular data types that you will is the object. We had a small run-in with the `object`, `Object`, and `{}` types, but never explained how you can pre-define an object.
+
+```ts
+function f(companyObj: { name: string }) {
+  console.log(companyObj.name);
+}
+const someObject = {
+  name: "Enki",
+  product: "EnkiApp"
+};
+f(someObject);
+```
+
+The function will only check that the `name` argument exist, and if it does it continues running.
+
+We can re-write this function using interfaces, to increase readability:
```

Comment:
If we wanted to reuse the type of `companyObj` at multiple places in our program, we can extract it into an interface: